coverage/artifacts/local.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package artifacts
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+//LocalArtifacts sub-type of Artifacts. Represent artifacts stored locally (
+// as oppose to artifacts stored in GCS bucket)
+type LocalArtifacts struct {
+	Artifacts
+}
+
+//NewLocalArtifacts constructs LocalArtifacts
+func NewLocalArtifacts(directory string, profileName string,
+	keyProfileName string, covStdoutName string) *LocalArtifacts {
+	return &LocalArtifacts{*New(
+		directory,
+		profileName,
+		keyProfileName,
+		covStdoutName)}
+}
+
+// ProfileReader creates and returns a ProfileReader by opening the file stored in profile path
+func (artifacts *LocalArtifacts) ProfileReader() (io.ReadCloser, error) {
```

Comment:
Not super sure we need this. Without this func, the caller can

```go
if f, err := os.Open(artifacts.ProfilePath()); err != nil {
    return fmt.Errorf("failed to open profile: %v", err)
}
```

With this func, you have

```go
if rc, err := artifacts.ProfileReader(); err != nil {
    return err
}
```

Said another way, when someone reads the code that calls `ProfileReader()` they'll need to dig in to this function to see what it's doing, whereas with the same amount of code but `os.Open()` it's clear.